src/dbnode/persist/schema/version_checker.go

```diff
@@ -0,0 +1,22 @@
+package schema
+
+// VersionChecker centralizes logic for checking if a major, minor version combo supports
+// specific functionality
+type VersionChecker struct {
+	majorVersion int
+	minorVersion int
+}
+
+func NewVersionChecker(majorVersion int, minorVersion int) *VersionChecker {
+	return &VersionChecker{
+		majorVersion: majorVersion,
+		minorVersion: minorVersion,
+	}
+}
+
+// IndexEntryValidationEnabled checks the version to determine if
+// fileset files of the specified version allow for doing checksum validation
+// on individual index entries
+func (v *VersionChecker) IndexEntryValidationEnabled() bool {
+	return v.majorVersion >= 1 && v.minorVersion >= 1
```

Comment:
Would the minor version reset to 0 the next major version bump? We’d still want to validate even after a major version bump I’m guessing?

Reply:
lol, great catch. This is wrong

Reply:
Fixed!